### PR TITLE
docs: Add migration guide for 6.x

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Migration Guide for 6.x

## :bulb: Motivation and Context

We want our users to be able to migrate easily to 6.x.

- [x] I reviewed the submitted code

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -1,6 +1,116 @@
...
+## Breaking Changes
+
+### Store Endpoint
+
+This version uses the [envelope endpoint](https://develop.sentry.dev/sdk/envelopes/).
+If you are using an on-premise installation it requires Sentry version
+`>= v20.6.0` to work. If you are using sentry.io nothing will change and
+no action is needed.
+
+### Sdk Inits
```
Comment (bruno-garcia) [inline MIGRATION.md:33]:
```suggestion
### SDK Inits
```

File: MIGRATION.md [hunk 2]
```diff
@@ -1,6 +1,116 @@
...
+In 5.x we use a nullable NSString to represent an event ID. The SDK, Hub and Client returned this
+nullable NSString for the event ID for capturing messages, events, errors, etc. With 6.x we have a new type SentryId which is not nullable to represent an event ID.
+Instead of returning `nil` when an event coulnd't be sent we return `SentryId.empty`.
+
+`5.x`
+
+```swift
+let eventId = SentrySDK.capture(message: "A message")
+if (nil != eventId) {
+    // event was sent
```
Comment (bruno-garcia) [inline MIGRATION.md:68]:
```suggestion
    // event was queued for submission
```

File: MIGRATION.md [hunk 3]
```diff
@@ -1,6 +1,116 @@
...
+    // event wasn't sent
+}
+```
+
+`6.x`
+
+```swift
+let eventId = SentrySDK.capture(message: "A message")
+if (eventId != SentryId.empty) {
+    // event was sent
```
Comment (bruno-garcia) [inline MIGRATION.md:88]:
```suggestion
    // event was queued for submission
```

File: MIGRATION.md [hunk 4]
```diff
@@ -1,6 +1,116 @@
...
+} else {
+    // event wasn't sent
+}
+```
+
+```objective-c
+SentryId *eventId = [SentrySDK captureMessage:@"A message"];
+if (eventId != SentryId.empty) {
+    // event was sent
```
Comment (bruno-garcia) [inline MIGRATION.md:97]:
```suggestion
    // event was queued for submission
```

**Review by bruno-garcia — APPROVED** [review]
Excellent. It's nice to see all the improvements (public facing) in one page.
A couple nits

**Comment by bruno-garcia** [comment]
We should surface this on the docs.
Maybe once it's done and approved we could move it to the docs and have it only there?
Here we could have a link to the docs.

**Comment by philipphofmann** [comment]
> We should surface this on the docs.
> Maybe once it's done and approved we could move it to the docs and have it only there?
> Here we could have a link to the docs.

Yep, that makes sense. I add a task to Asana, so we don't forget it for next week.

**Pull request merged**